packages/transformers/js/core/src/collect.rs

```diff
@@ -767,6 +767,25 @@ impl Visit for Collect {
           self.used_imports.insert(id!(ident));
         }
       }
+      Expr::Bin(bin_expr) => {
+        // Some TSC polyfills use a pattern like below.
+        // We want to avoid marking these modules as CJS
+        // e.g. var _polyfill = (this && this.polyfill) || function () {}
+        if matches!(*bin_expr.left, Expr::This(..)) {
```

Comment:
only do this if we are `in_module_this`, and also check that the operator is `&&` to be safe?

Reply:
> only do this if we are in_module_this
Ohh I meant to scope it to `in_module_this` 🤦 

> check that the operator is && to be safe
👍

Reply:
@devongovett Done